contracts/TokenRequest.sol

```diff
@@ -0,0 +1,136 @@
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "./lib/ArrayUtils.sol";
+
+
+contract TokenRequest is AragonApp {
+    using SafeMath for uint256;
+
+    bytes32 constant public TOKEN_REQUEST_ROLE = keccak256("TOKEN_REQUEST_ROLE");
+    bytes32 constant public REFOUND_ROLE = keccak256("REFOUND_ROLE");
+    bytes32 constant public VOTING_TOKEN_REQUEST_ROLE = keccak256("VOTING_TOKEN_REQUEST_ROLE");
+
+    string private constant ERROR_VAULT_NOT_CONTRACT = "TOKENREQUEST_VAULT_NOT_CONTRACT";
+    string private constant ERROR_TOKEN_MANAGER_NOT_CONTRACT = "TOKENREQUEST_TOKEN_MANAGER_NOT_CONTRACT";
+    string private constant ERROR_TOKEN_NOT_CONTRACT = "TOKENREQUEST_TOKEN_NOT_CONTRACT";
+    string private constant ERROR_CANNOT_MINT_ZERO = "TOKENREQUEST_CANNOT_MINT_ZERO";
+    string private constant ERROR_ETH_VALUE_MISMATCH = "TOKENREQUEST_ETH_VALUE_MISMATCH";
+    string private constant ERROR_DEPOSIT_VALUE_MISMATCH = "TOKENREQUEST_DEPOSIT_VALUE_MISMATCH";
+    string private constant ERROR_DEPOSIT_TOKEN_MISMATCH = "TOKENREQUEST_DEPOSIT_TOKEN_MISMATCH";
+    string private constant ERROR_MINT_VALUE_MISMATCH = "TOKENREQUEST_MINT_VALUE_MISMATCH";
+    string private constant ERROR_DEPOSIT_NOT_ACTIVE = "TOKENREQUEST_DEPOSIT_NOT_ACTIVE";
+    string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKENREQUEST_TOKEN_TRANSFER_REVERT";
+    string private constant ERROR_TOKEN_APPROVE_FAILED = "TOKENREQUEST_TKN_APPROVE_FAILED";
+
+     struct Request {
+        uint256 depositAmount;
+        address token;
+        uint256 mintAmount;
+        bool active;
+    }
+
+    Vault public vault;
+    TokenManager public tokenManager;
+    Voting public voting;
+
+    // In case of refound how can we know which pending refound to do?
+    mapping (address => mapping (uint256 => Request)) internal requests;
+    mapping (address => uint256) public requestsLengths;
+
+    event Request(address indexed receiver, address depositToken, uint256 depositAmount, uint256 mintAmount, uint256 voteId, uint256 requestId);
+    event ApprovedRequest(
+        address indexed receiver,
+        address depositToken,
+        uint256 depositAmount,
+        uint256 mintAmount,
+        uint256 voteId,
+        uint256 requestId
+        );
+
+     /**
+    * @notice Initialize TokenRequest app contract
+    * @param _vault Address of the vault
+    * @param _tokenManager TokenManager address
+    */
+    function initialize(Vault _vault, TokenManager _tokenManager) external onlyInit {
+        initialized();
+
+        require(isContract(_vault), ERROR_VAULT_NOT_CONTRACT);
+        require(isContract(_tokenManager), ERROR_TOKEN_MANAGER_NOT_CONTRACT);
+
+        vault = _vault;
+        tokenManager = _tokenManager;
+    }
+
+    function request(
+        address _depositToken,
+        uint256 _depositAmount,
+        uint256 _mintAmount,
+        bytes _evmScript,
```

Comment:
Could we use `_executionScript` to follow the same standard as the rest of the apps?

Reply:
you mean the variable name? yes sure!